### PR TITLE
WIP Temporary fix of the TransportSolver FV scheme to improve the performance of some jupyter notebook demos

### PR DESCRIPTION
**Pull request by mtsveta**
Fully-implicit finite volume scheme is added to fix the TransportSolver

### CODE REVIEW

**Comment by mtsveta** [comment]
Hi, Allan! This PR is important ONLY IF we are not merging the whole `cpp-reactivetransportsolver-demo` branch to `master` soon. I have told you that there is a little issue with the FV scheme in the ReactiveTransport, which makes one jupyter-notebook tutorial to produce slightly disturbed curves during reactive transport simulations. Generally, we will also need to create a new conda image for reaktoro to get rid of this issue. So maybe, making a bigger merge (with an updated version of SmartEquilibriumSolver) would be evn better.

File: Reaktoro/Transport/TransportSolver.hpp [hunk 1]
```diff
@@ -204,7 +204,7 @@ class Mesh
 };
 
 /// A class for solving advection-diffusion problem.
-/// Eq: du/dt + v*du/dx = D*d�u/dx�
+/// Eq: du/dt + v*du/dx = D*d�u/dx�
```
Comment (allanleal) [inline Reaktoro/Transport/TransportSolver.hpp]:
```suggestion
/// Eq: du/dt + v*du/dx = D*d2u/dx2
```

File: Reaktoro/Transport/TransportSolver.cpp [hunk 2]
```diff
@@ -318,7 +342,7 @@ auto TransportSolver::step(VectorRef u, VectorConstRef q) -> void
     u[icell0] += aux * alpha * (ul - u0[0]) + (3.0*diffusion*ul*dt/(dx*dx)); // prescribed amount on the wall and approximatin deriveted by forward diference approximation with second order error
 
     // Handle the right boundary cell
-    u[icelln] += alpha * (u0[icelln - 1] - u0[icelln]); // du/dx = 0 at the right boundary
+    u[icelln] += alpha * (u0[icelln - 1] - u0[icelln]); // du/dx = 0 at the right boundary*/
```
Comment (allanleal) [inline Reaktoro/Transport/TransportSolver.cpp]: Please check `*/`.

File: Reaktoro/Transport/TransportSolver.cpp [hunk 3]
```diff
@@ -249,26 +249,44 @@ TransportSolver::TransportSolver()
...
+    const auto beta = velocity*dt/dx;
     const auto num_cells = mesh_.numCells();
     const auto icell0 = 0;
     const auto icelln = num_cells - 1;
 
     A.resize(num_cells);
     phi.resize(num_cells);
 
+    // Initialize coefficients of the system's matrix
+    double a(0.0), b(0.0), c(0.0);
```
Comment (allanleal) [inline Reaktoro/Transport/TransportSolver.cpp]:
```suggestion
    double a = 0.0, b = 0.0, c = 0.0;
```

**Comment by allanleal** [comment]
Hi Svetlana, let's replicate this PR on Reaktoro v2 when we bring the reactive transport functionality back.

**Pull request closed without merging**